lib/auth.js

```diff
@@ -81,13 +81,21 @@ function useProvideAuth() {
   };
 }
 
-const formatUser = (user) => {
+const getStripeRole = async () => {
+  await firebase.auth().currentUser.getIdToken(true);
+  const decodedToken = await firebase.auth().currentUser.getIdTokenResult();
+
+  return decodedToken.claims.stripeRole || 'free';
+};
+
+const formatUser = async (user) => {
   return {
     uid: user.uid,
     email: user.email,
     name: user.displayName,
     token: user.xa,
     provider: user.providerData[0].providerId,
-    photoUrl: user.photoURL
+    photoUrl: user.photoURL,
+    stripeRole: await getStripeRole()
```

Comment:
Just a heads up for anyone else who had problems accessing `.xa` to get the token, you should be able to use this approach to get both the `token` and `stripeRole`:

```js
const formatUser = async (user) => {
  const { token, claims } = await user.getIdTokenResult();

  return {
    uid: user.uid,
    email: user.email,
    name: user.displayName,
    token,
    provider: user.providerData[0].providerId,
    photoUrl: user.photoURL,
    stripeRole: claims.stripeRole || "free",
  };
};
```

Reply:
Related to this? https://github.com/leerob/fastfeedback/issues/25

Reply:
Yes! Just extending it to include the Stripe role too